<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AddSlaveDialog</class>
 <widget class="QDialog" name="AddSlaveDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>545</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="styleSheet">
   <string notr="true">* {
  font: 13pt;
}

QLabel#addrWarningLabel,#typeWarningLabel,#nameWarningLabel,#stateWarningLabel {
  font: 9pt;
  color: red;
}</string>
  </property>
  <layout class="QVBoxLayout" name="widgetLayout">
   <item>
    <layout class="QHBoxLayout" name="centralLayout">
     <item>
      <widget class="QWidget" name="leftWidget" native="true">
       <property name="minimumSize">
        <size>
         <width>216</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>216</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="leftLayout">
        <property name="spacing">
         <number>10</number>
        </property>
        <property name="leftMargin">
         <number>6</number>
        </property>
        <property name="topMargin">
         <number>10</number>
        </property>
        <property name="rightMargin">
         <number>10</number>
        </property>
        <property name="bottomMargin">
         <number>10</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="lButtonLayout">
          <item>
           <widget class="QPushButton" name="spacePushButton">
            <property name="text">
             <string>刷新</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="resetPushButton">
            <property name="text">
             <string>重置</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="addrLayout">
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="addrLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>地  址：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="addrLineEdit">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string/>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
            <property name="readOnly">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="addrWarningLabel">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>地址必须是十六进制数取值范围：00~EF</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="typeLayout">
          <item>
           <widget class="QLabel" name="typeLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>类  型：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="typeLineEdit">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
            <property name="readOnly">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="typeWarningLabel">
          <property name="text">
           <string>类型必须是长度小于等于2的字母或数字</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="nameLayout">
          <item>
           <widget class="QLabel" name="nameLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>名  称：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="nameLineEdit">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
            <property name="readOnly">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="nameWarningLabel">
          <property name="text">
           <string>名称必须是长度小于等于8的字母或数字</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="memoLayout">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="memoLineLayout">
            <property name="spacing">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="memoLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>备  注</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="Line" name="memoLine">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="memoPlainTextEdit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="lineWrapMode">
             <enum>QPlainTextEdit::WidgetWidth</enum>
            </property>
            <property name="readOnly">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="Line" name="centralLine">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="rightWidget" native="true">
       <property name="minimumSize">
        <size>
         <width>304</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>304</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="rightLayout">
        <property name="spacing">
         <number>10</number>
        </property>
        <property name="leftMargin">
         <number>8</number>
        </property>
        <property name="topMargin">
         <number>10</number>
        </property>
        <property name="rightMargin">
         <number>6</number>
        </property>
        <property name="bottomMargin">
         <number>10</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="labelLayout">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <widget class="Line" name="labelLLine">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelMLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>功  能  列  表</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="labelRLine">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="rButtonLayout">
          <item>
           <layout class="QHBoxLayout" name="rDButtonLayout">
            <item>
             <widget class="QPushButton" name="addRowPushButton">
              <property name="text">
               <string>添加行</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="insertRowPushButton">
              <property name="text">
               <string>插入行</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="delRowPushButton">
            <property name="text">
             <string>删除当前行</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableWidget" name="stateTableWidget">
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectItems</enum>
          </property>
          <property name="rowCount">
           <number>1</number>
          </property>
          <attribute name="horizontalHeaderCascadingSectionResizes">
           <bool>false</bool>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>true</bool>
          </attribute>
          <row/>
          <column>
           <property name="text">
            <string> 状态字 </string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>状态信息</string>
           </property>
          </column>
          <item row="0" column="0">
           <property name="text">
            <string>00</string>
           </property>
           <property name="textAlignment">
            <set>AlignCenter</set>
           </property>
           <property name="flags">
            <set>ItemIsSelectable|ItemIsEnabled</set>
           </property>
          </item>
          <item row="0" column="1">
           <property name="text">
            <string>关闭</string>
           </property>
           <property name="textAlignment">
            <set>AlignCenter</set>
           </property>
           <property name="flags">
            <set>ItemIsSelectable|ItemIsEnabled</set>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="stateWarningLabel">
          <property name="text">
           <string>不可空项、状态字(00~FF)不可重复、状态信息长度小于8</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>AddSlaveDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>AddSlaveDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
